sqlglot/optimizer/annotate_types.py

```diff
@@ -287,6 +287,7 @@ class TypeAnnotator(metaclass=_TypeAnnotator):
         exp.Max: lambda self, e: self._annotate_by_args(e, "this", "expressions"),
         exp.Min: lambda self, e: self._annotate_by_args(e, "this", "expressions"),
         exp.Null: lambda self, e: self._annotate_with_type(e, exp.DataType.Type.NULL),
+        exp.Slice: lambda self, e: self._annotate_with_type(e, exp.DataType.Type.UNKNOWN),
```

Comment:
shouldn’t alice be i’m the t of list <t>

Reply:
Ah due to unknown

Reply:
can’t slice type be inferred if the underlying obj is an list<t>

Reply:
I think slice corresponds to the "range" object and the bracket is what you mean? Not sure if it makes sense to assign a type to 1:3 for example, but a[1:3] can have eg a type of array of ints.

Reply:
ah ok

Reply:
I'm testing the subarray type we discussed now, might need to make a few small changes to fix it too